src/pkg/services/m365/api/lists.go

```diff
@@ -251,42 +251,49 @@ func (c Lists) PostList(
 		BySiteId(siteID).
 		Lists().
 		Post(ctx, newList, nil)
+	if err != nil {
+		return nil, graph.Wrap(ctx, err, "creating list")
+	}
+
+	listItems, err := c.PostListItems(ctx, siteID, restoredList)
+	if err == nil {
+		restoredList.SetItems(listItems)
+		return restoredList, nil
+	}
 
-	return restoredList, graph.Wrap(ctx, err, "creating list").OrNil()
+	// rollback list creation
+	err = c.DeleteList(ctx, siteID, ptr.Val(restoredList.GetId()))
+
+	return nil, graph.Wrap(ctx, err, "deleting restored list after items creation failure").OrNil()
 }
 
-func (c Lists) PostListItem(
+func (c Lists) PostListItems(
 	ctx context.Context,
-	siteID, listID string,
-	oldListByteArray []byte,
+	siteID string,
+	restoredList models.Listable,
 ) ([]models.ListItemable, error) {
-	oldList, err := BytesToListable(oldListByteArray)
-	if err != nil {
-		return nil, clues.WrapWC(ctx, err, "generating list item from stored bytes")
-	}
+	listItems := make([]models.ListItemable, 0)
 
-	contents := make([]models.ListItemable, 0)
-
-	for _, itm := range oldList.GetItems() {
+	for _, itm := range restoredList.GetItems() {
 		temp := CloneListItem(itm)
```

Comment:
hmm, can you please explain why cloning the item is necessary?

Reply:
Instead of directly passing all the attributes from the `storedListItemBytes`, we check and filter them to pass only the attributes and data enough to create the list item.

Else we might encounter errors while creation of list item:
- 'Invalid request' --> if CreatedByUser / LastModifiedByUser is set with nil
- 'Invalid request' --> if number of columns do not match number of field data of the list item
- 'ColumnDefinition unidentifiable' --> In cases of certain columns like location, there is no way to identify if is a location column or not. So we default to text instead of not sending any column. The address value is then populated as text while list item creation.
- 'General exception while processing' --> if DefaultValue and Validation attributes of models.ColumnDefinitionable is set with nil